enhancements/monitoring/multi-tenant-alerting.md

```diff
@@ -0,0 +1,494 @@
+    receiver: default-user-receiver
+receivers:
+- name: default-platform-receiver
+  ...
+- name: default-user-receiver
+```
+
+Note that a similar use case was already reported in [BZ 1933239][bz-1933239].
+
+
+#### Tenancy
+
+By design, all alerts coming from UWM have a `namespace` label equal to the
+`PrometheusRule` resource's namespace. The Prometheus operator relies on this
+invariant to generate an Alertmanager configuration that ensures that a given
+`AlertmanagerConfig` resource only matches alerts that have the same
+`namespace` value. This means that an `AlertmanagerConfig` resource from
+namespace `foo` only processes alerts with the `namespace="foo"` label (be it
+for routing or inhibiting purposes).
+
+Below is how the operator renders an AlertmanagerConfig resource in the final Alertmanager configuration.
+
+```yaml
+...
+route:
+  routes:
+  # The next item is generated from an AlertmanagerConfig resource named alertmanagerconfig1 in namespace foo.
+  - matchers: ['namespace="foo"']
+    receiver: foo-alertmanagerconfig1-my-receiver
+    continue: true
+    routes:
+    - ...
+inhibit_rules:
+# The next item is generated from an AlertmanagerConfig resource named alertmanagerconfig1 in namespace foo.
+- source_matchers: ['namespace="foo"', ...]
+  target_maTCHERS: ['namespace="foo"', ...]
+  equal: ['namespace', ...]
+receivers:
+# The next item is generated from an AlertmanagerConfig resource named alertmanagerconfig1 in namespace foo.
+- name: foo-alertmanagerconfig1-my-receiver
+  ...
+```
+
+#### RBAC
+
+The cluster monitoring operator ships a new cluster role
+`alertmanager-config-edit` so that cluster admins can bind it with a
+`RoleBinding` to grant permissions to users or groups on `AlertmanagerConfig`
+custom resources within a given namespace.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: alertmanager-config-edit
+rules:
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - alertmanagerconfigs
+  verbs:
+  - '*'
+```
+
+This role complements the existing `monitoring-edit`, `monitoring-rules-edit` and `monitoring-rules-view` roles.
+
+#### Resource impacts
+
+The size of the Alertmanager routing tree can grow to thousands of entries
+(possibly several `AlertmanagerConfig` resources per namespace) and this may
+hinder the performances of the Plaform Alertmanager (increased latency of alert
+notifications for instance).
+
+We already know from upstream users that Alertmanager can deal with many
+routes. We plan to simulate environments with thousands of `AlertmanagerConfig`
+resources and measure the impact on notification delivery.
+
+### Risks and Mitigations
+
+#### Disruption of the platform Alertmanager
+
+Even though the Prometheus operator prevents it as much as it can, it may be
+possible for users to create an `AlertmanagerConfig` resource that triggers the
+Prometheus operator to generate an invalid Alertmanager configuration, leading
```

Comment:
The operator will do its best to reject invalid configuration but bugs happen and it may happen that the operator generates a configuration that couldn't be parsed by Alertmanager. In short, unlikely to happen but still possible.